### PR TITLE
Add RPC methods for querying validator status

### PR DESCRIPTION
**Pull request by sisou**
## What's in this pull request?

Add RPC methods `getIsValidatorElected() => bool` and `getIsValidatorSynced() => bool` to enable querying this information for running validators. Otherwise, this information is only available by parsing logs.

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: rpc-server/src/dispatchers/validator.rs [hunk 1]
```diff
@@ -57,4 +57,15 @@ impl ValidatorInterface for ValidatorDispatcher {
         log::debug!("Automatic reactivation set to {}.", automatic_reactivate);
         Ok(().into())
     }
+
+    async fn get_is_validator_elected(&mut self) -> RPCResult<bool, (), Self::Error> {
```
Comment (hrxi) [inline rpc-server/src/dispatchers/validator.rs]: `get_is_…` sounds a bit weird to me, I'd prefer `is_…`. Looking at the functions we already expose in the RPC, I can see a couple of both variants.

Comment (sisou) [reply rpc-server/src/dispatchers/validator.rs]: @viquezclaudio please decide, or find someone to decide which naming convention to use.

Comment (hrxi) [reply rpc-server/src/dispatchers/validator.rs]: We briefly talked about it, and decided to rename everything to `is_…`: #2181.

Comment (jsdanielh) [reply rpc-server/src/dispatchers/validator.rs]: @sisou Could you make the change and rebase the branch?

Comment (sisou) [reply rpc-server/src/dispatchers/validator.rs]: Done :+1:

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**